daisybill/github_workflow
# Support for Git Worktrees

## Problem

- I use [git-worktrees](https://git-scm.com/docs/git-worktree) to work on features concurrently without needing to clone another copy of a repo (check it out, it's awesome)
- I can't have the branch, `main`, checked out in two worktrees attached to the same repo, so when I use github_workflow from the worktree I'm using to build the feature I'm working on and I have another worktree attached to the same repo with `main` checked out, then github_workflow breaks because it can't checkout `main` on the feature branch I'm working on.

## Proposed Solution

- This PR makes pulling, rebasing, and creating branches a little safer by removing the need to have to checkout to the `main` branch before doing those operations
- This way, github_workflow works the same as it did before, but now also support worktrees

## Testing

###  Test Case 1

- Create a git worktree and cd into it
- use github_workflow as you normally would
- Expect it to fail before these changes

### Test Case 2

- build and install gem from pr/branch/commit
- ensure existing github_workflow works as it normally does
- Expect no breaking changes

### Test Case 3

- build and install gem from pr/branch/commit
- create a git worktree and cd into it
- use github_workflow as you normally would
- Expect to not see failures


## Misc

- I upgraded and tested with ruby 2.7.2
- I am using this version of the gem in my workflow for testing
